
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://Jianhua-Wang.github.io/mafm/tutorial/multi-input/">
      
      
        <link rel="prev" href="../single-input/">
      
      
        <link rel="next" href="../web-visualization/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.14">
    
    
      
        <title>Multi-Input Fine-Mapping - mafm</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.342714a4.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/_mkdocstrings.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="preference" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#multi-input-fine-mapping" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="mafm" class="md-header__button md-logo" aria-label="mafm" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            mafm
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Multi-Input Fine-Mapping
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/Jianhua-Wang/mafm" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    Jianhua-Wang/mafm
  </div>
</a>
      </div>
    
  </nav>
  
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="mafm" class="md-nav__button md-logo" aria-label="mafm" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    mafm
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/Jianhua-Wang/mafm" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    Jianhua-Wang/mafm
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../installation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Installation
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../usage.md" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usage
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" checked>
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    Tutorial
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Tutorial
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../getting-started/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Getting Started
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../quick-start/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Quick Start
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../single-input/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Single-Input Fine-Mapping
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Multi-Input Fine-Mapping
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Multi-Input Fine-Mapping
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#when-to-use-multi-input-strategy" class="md-nav__link">
    <span class="md-ellipsis">
      When to Use Multi-Input Strategy
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#multi-input-workflow-components" class="md-nav__link">
    <span class="md-ellipsis">
      Multi-Input Workflow Components
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Multi-Input Workflow Components">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-meta-analysis-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      1. Meta-Analysis Strategies
    </span>
  </a>
  
    <nav class="md-nav" aria-label="1. Meta-Analysis Strategies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#cross-ancestry-meta-analysis-meta_all" class="md-nav__link">
    <span class="md-ellipsis">
      Cross-Ancestry Meta-Analysis (meta_all)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#within-ancestry-meta-analysis-meta_by_population" class="md-nav__link">
    <span class="md-ellipsis">
      Within-Ancestry Meta-Analysis (meta_by_population)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#no-meta-analysis-no_meta" class="md-nav__link">
    <span class="md-ellipsis">
      No Meta-Analysis (no_meta)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-quality-control-across-studies" class="md-nav__link">
    <span class="md-ellipsis">
      2. Quality Control Across Studies
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. Quality Control Across Studies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#cross-study-qc-metrics" class="md-nav__link">
    <span class="md-ellipsis">
      Cross-Study QC Metrics
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-multi-input-fine-mapping-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      3. Multi-Input Fine-Mapping Strategies
    </span>
  </a>
  
    <nav class="md-nav" aria-label="3. Multi-Input Fine-Mapping Strategies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#a-multi-input-tools-multi_input" class="md-nav__link">
    <span class="md-ellipsis">
      A. Multi-Input Tools (multi_input)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#b-post-hoc-combination-post_hoc_combine" class="md-nav__link">
    <span class="md-ellipsis">
      B. Post-hoc Combination (post_hoc_combine)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#detailed-tool-descriptions" class="md-nav__link">
    <span class="md-ellipsis">
      Detailed Tool Descriptions
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Detailed Tool Descriptions">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#multisusie" class="md-nav__link">
    <span class="md-ellipsis">
      MultiSuSiE
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#susiex" class="md-nav__link">
    <span class="md-ellipsis">
      SuSiEx
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#comprehensive-multi-input-example" class="md-nav__link">
    <span class="md-ellipsis">
      Comprehensive Multi-Input Example
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Comprehensive Multi-Input Example">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-input-preparation" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Input Preparation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-quality-control" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Quality Control
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3a-cross-ancestry-analysis-with-multisusie" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3A: Cross-Ancestry Analysis with MultiSuSiE
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3b-population-specific-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3B: Population-Specific Analysis
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-compare-approaches" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Compare Approaches
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#advanced-multi-input-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      Advanced Multi-Input Considerations
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Advanced Multi-Input Considerations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#handling-missing-data" class="md-nav__link">
    <span class="md-ellipsis">
      Handling Missing Data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#population-stratification" class="md-nav__link">
    <span class="md-ellipsis">
      Population Stratification
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#computational-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      Computational Considerations
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#interpreting-multi-input-results" class="md-nav__link">
    <span class="md-ellipsis">
      Interpreting Multi-Input Results
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Interpreting Multi-Input Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#population-specific-effects" class="md-nav__link">
    <span class="md-ellipsis">
      Population-Specific Effects
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shared-causal-variants" class="md-nav__link">
    <span class="md-ellipsis">
      Shared Causal Variants
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#meta-analysis-benefits" class="md-nav__link">
    <span class="md-ellipsis">
      Meta-Analysis Benefits
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#common-multi-input-issues" class="md-nav__link">
    <span class="md-ellipsis">
      Common Multi-Input Issues
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next-steps" class="md-nav__link">
    <span class="md-ellipsis">
      Next Steps
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../web-visualization/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Web Visualization
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../web-example/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Web Example
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../advanced/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Advanced Topics
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../api/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Modules
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../changelog/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Changelog
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#when-to-use-multi-input-strategy" class="md-nav__link">
    <span class="md-ellipsis">
      When to Use Multi-Input Strategy
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#multi-input-workflow-components" class="md-nav__link">
    <span class="md-ellipsis">
      Multi-Input Workflow Components
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Multi-Input Workflow Components">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-meta-analysis-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      1. Meta-Analysis Strategies
    </span>
  </a>
  
    <nav class="md-nav" aria-label="1. Meta-Analysis Strategies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#cross-ancestry-meta-analysis-meta_all" class="md-nav__link">
    <span class="md-ellipsis">
      Cross-Ancestry Meta-Analysis (meta_all)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#within-ancestry-meta-analysis-meta_by_population" class="md-nav__link">
    <span class="md-ellipsis">
      Within-Ancestry Meta-Analysis (meta_by_population)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#no-meta-analysis-no_meta" class="md-nav__link">
    <span class="md-ellipsis">
      No Meta-Analysis (no_meta)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-quality-control-across-studies" class="md-nav__link">
    <span class="md-ellipsis">
      2. Quality Control Across Studies
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. Quality Control Across Studies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#cross-study-qc-metrics" class="md-nav__link">
    <span class="md-ellipsis">
      Cross-Study QC Metrics
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-multi-input-fine-mapping-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      3. Multi-Input Fine-Mapping Strategies
    </span>
  </a>
  
    <nav class="md-nav" aria-label="3. Multi-Input Fine-Mapping Strategies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#a-multi-input-tools-multi_input" class="md-nav__link">
    <span class="md-ellipsis">
      A. Multi-Input Tools (multi_input)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#b-post-hoc-combination-post_hoc_combine" class="md-nav__link">
    <span class="md-ellipsis">
      B. Post-hoc Combination (post_hoc_combine)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#detailed-tool-descriptions" class="md-nav__link">
    <span class="md-ellipsis">
      Detailed Tool Descriptions
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Detailed Tool Descriptions">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#multisusie" class="md-nav__link">
    <span class="md-ellipsis">
      MultiSuSiE
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#susiex" class="md-nav__link">
    <span class="md-ellipsis">
      SuSiEx
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#comprehensive-multi-input-example" class="md-nav__link">
    <span class="md-ellipsis">
      Comprehensive Multi-Input Example
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Comprehensive Multi-Input Example">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-input-preparation" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Input Preparation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-quality-control" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Quality Control
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3a-cross-ancestry-analysis-with-multisusie" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3A: Cross-Ancestry Analysis with MultiSuSiE
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3b-population-specific-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3B: Population-Specific Analysis
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-compare-approaches" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Compare Approaches
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#advanced-multi-input-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      Advanced Multi-Input Considerations
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Advanced Multi-Input Considerations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#handling-missing-data" class="md-nav__link">
    <span class="md-ellipsis">
      Handling Missing Data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#population-stratification" class="md-nav__link">
    <span class="md-ellipsis">
      Population Stratification
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#computational-considerations" class="md-nav__link">
    <span class="md-ellipsis">
      Computational Considerations
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#interpreting-multi-input-results" class="md-nav__link">
    <span class="md-ellipsis">
      Interpreting Multi-Input Results
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Interpreting Multi-Input Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#population-specific-effects" class="md-nav__link">
    <span class="md-ellipsis">
      Population-Specific Effects
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shared-causal-variants" class="md-nav__link">
    <span class="md-ellipsis">
      Shared Causal Variants
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#meta-analysis-benefits" class="md-nav__link">
    <span class="md-ellipsis">
      Meta-Analysis Benefits
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#common-multi-input-issues" class="md-nav__link">
    <span class="md-ellipsis">
      Common Multi-Input Issues
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next-steps" class="md-nav__link">
    <span class="md-ellipsis">
      Next Steps
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="multi-input-fine-mapping">Multi-Input Fine-Mapping<a class="headerlink" href="#multi-input-fine-mapping" title="Permanent link">&para;</a></h1>
<p>Multi-input fine-mapping analyzes multiple cohorts and ancestries simultaneously to leverage shared genetic architecture while accounting for population differences. This is where MAFM truly shines, offering sophisticated approaches to multi-ancestry genetic analysis.</p>
<h2 id="when-to-use-multi-input-strategy">When to Use Multi-Input Strategy<a class="headerlink" href="#when-to-use-multi-input-strategy" title="Permanent link">&para;</a></h2>
<div class="admonition tip">
<p class="admonition-title">Multi-Input Use Cases</p>
<dl>
<dt><strong>Multiple ancestries available</strong></dt>
<dd>Leverage power across populations while modeling LD differences</dd>
<dt><strong>Shared causal architecture</strong></dt>
<dd>When you expect similar causal variants across populations</dd>
<dt><strong>Increased statistical power</strong></dt>
<dd>Combine sample sizes for improved fine-mapping resolution</dd>
<dt><strong>Cross-population validation</strong></dt>
<dd>Identify variants with consistent effects across ancestries</dd>
</dl>
</div>
<h2 id="multi-input-workflow-components">Multi-Input Workflow Components<a class="headerlink" href="#multi-input-workflow-components" title="Permanent link">&para;</a></h2>
<h3 id="1-meta-analysis-strategies">1. Meta-Analysis Strategies<a class="headerlink" href="#1-meta-analysis-strategies" title="Permanent link">&para;</a></h3>
<p>The first step is deciding how to combine your studies:</p>
<h4 id="cross-ancestry-meta-analysis-meta_all">Cross-Ancestry Meta-Analysis (<code>meta_all</code>)<a class="headerlink" href="#cross-ancestry-meta-analysis-meta_all" title="Permanent link">&para;</a></h4>
<p>Combines all studies regardless of ancestry:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>mafm<span class="w"> </span>meta<span class="w"> </span>input_loci.txt<span class="w"> </span>meta_output/<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="w">    </span>--meta-method<span class="w"> </span>meta_all<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a><span class="w">    </span>--threads<span class="w"> </span><span class="m">4</span>
</code></pre></div>
<p><strong>When to use:</strong>
- Strong prior belief in shared causal variants
- Large effect sizes relative to population differences
- Increased power is the primary goal</p>
<p><strong>What it does:</strong>
- Performs inverse-variance weighted meta-analysis of summary statistics
- Sample-size weighted averaging of LD matrices
- Creates single meta-analyzed dataset per locus</p>
<h4 id="within-ancestry-meta-analysis-meta_by_population">Within-Ancestry Meta-Analysis (<code>meta_by_population</code>)<a class="headerlink" href="#within-ancestry-meta-analysis-meta_by_population" title="Permanent link">&para;</a></h4>
<p>Combines studies within each ancestry separately:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>mafm<span class="w"> </span>meta<span class="w"> </span>input_loci.txt<span class="w"> </span>meta_output/<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a><span class="w">    </span>--meta-method<span class="w"> </span>meta_by_population<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a><span class="w">    </span>--threads<span class="w"> </span><span class="m">4</span>
</code></pre></div>
<p><strong>When to use:</strong>
- Population-specific effect sizes expected
- Want to preserve ancestry-specific signals
- Balanced approach between power and specificity</p>
<p><strong>What it does:</strong>
- Meta-analyzes EUR studies together, AFR studies together, etc.
- Maintains population-specific LD structure
- Creates separate datasets for each ancestry</p>
<h4 id="no-meta-analysis-no_meta">No Meta-Analysis (<code>no_meta</code>)<a class="headerlink" href="#no-meta-analysis-no_meta" title="Permanent link">&para;</a></h4>
<p>Keeps all studies separate:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>mafm<span class="w"> </span>meta<span class="w"> </span>input_loci.txt<span class="w"> </span>meta_output/<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a><span class="w">    </span>--meta-method<span class="w"> </span>no_meta<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a><span class="w">    </span>--threads<span class="w"> </span><span class="m">4</span>
</code></pre></div>
<p><strong>When to use:</strong>
- Highly heterogeneous effect sizes
- Study-specific technical factors
- Maximum preservation of individual study characteristics</p>
<h3 id="2-quality-control-across-studies">2. Quality Control Across Studies<a class="headerlink" href="#2-quality-control-across-studies" title="Permanent link">&para;</a></h3>
<p>Multi-study QC provides insights impossible with single studies:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>mafm<span class="w"> </span>qc<span class="w"> </span>input_loci.txt<span class="w"> </span>qc_output/<span class="w"> </span>--threads<span class="w"> </span><span class="m">4</span>
</code></pre></div>
<h4 id="cross-study-qc-metrics">Cross-Study QC Metrics<a class="headerlink" href="#cross-study-qc-metrics" title="Permanent link">&para;</a></h4>
<dl>
<dt><strong>Cochran's Q Test</strong></dt>
<dd>Tests for heterogeneity in effect sizes across studies</dd>
<dd>High Q values suggest population or study-specific effects</dd>
<dt><strong>SNP Missingness Patterns</strong></dt>
<dd>Identifies variants missing in specific populations</dd>
<dd>Helps understand coverage differences across ancestries</dd>
<dt><strong>LD Structure Comparison</strong></dt>
<dd>Compares LD patterns across populations</dd>
<dd>Identifies regions with dramatically different LD structure</dd>
<dt><strong>Cross-Ancestry MAF Correlations</strong></dt>
<dd>Compares allele frequencies across populations</dd>
<dd>Detects potential population stratification or technical issues</dd>
</dl>
<div class="admonition info">
<p class="admonition-title">QC Output Files</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>qc_output/
<a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>├── cochran_q.txt           # Effect size heterogeneity
<a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>├── snp_missingness.txt     # Coverage patterns
<a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a>├── ld_4th_moment.txt       # LD structure comparison
<a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>├── s_estimate.txt          # Consistency parameters
<a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a>└── maf_comparison.txt      # Frequency comparisons
</code></pre></div>
</div>
<h3 id="3-multi-input-fine-mapping-strategies">3. Multi-Input Fine-Mapping Strategies<a class="headerlink" href="#3-multi-input-fine-mapping-strategies" title="Permanent link">&para;</a></h3>
<p>MAFM offers three strategies for multi-input fine-mapping:</p>
<h4 id="a-multi-input-tools-multi_input">A. Multi-Input Tools (<code>multi_input</code>)<a class="headerlink" href="#a-multi-input-tools-multi_input" title="Permanent link">&para;</a></h4>
<p>Use tools specifically designed for multi-population analysis:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>mafm<span class="w"> </span>finemap<span class="w"> </span>input_loci.txt<span class="w"> </span>output/<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a><span class="w">    </span>--strategy<span class="w"> </span>multi_input<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a><span class="w">    </span>--tool<span class="w"> </span>multisusie<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a><span class="w">    </span>--max-causal<span class="w"> </span><span class="m">5</span>
</code></pre></div>
<p><strong>Supported Tools:</strong></p>
<ul>
<li><strong>MultiSuSiE</strong>: Multi-population extension of SuSiE</li>
<li><strong>SuSiEx</strong>: Cross-ancestry fine-mapping tool</li>
</ul>
<h4 id="b-post-hoc-combination-post_hoc_combine">B. Post-hoc Combination (<code>post_hoc_combine</code>)<a class="headerlink" href="#b-post-hoc-combination-post_hoc_combine" title="Permanent link">&para;</a></h4>
<p>Run single-input tools on each study, then intelligently combine results:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>mafm<span class="w"> </span>finemap<span class="w"> </span>input_loci.txt<span class="w"> </span>output/<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a><span class="w">    </span>--strategy<span class="w"> </span>post_hoc_combine<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a><span class="w">    </span>--tool<span class="w"> </span>susie<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a><span class="w">    </span>--combine-cred<span class="w"> </span>union<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a><span class="w">    </span>--combine-pip<span class="w"> </span>max<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-6-6" name="__codelineno-6-6" href="#__codelineno-6-6"></a><span class="w">    </span>--jaccard-threshold<span class="w"> </span><span class="m">0</span>.1
</code></pre></div>
<p><strong>Combination Methods:</strong></p>
<div class="admonition note">
<p class="admonition-title">Credible Set Combination (<code>--combine-cred</code>)</p>
<dl>
<dt><strong><code>union</code></strong> (default)</dt>
<dd>Union of all credible sets from individual studies</dd>
<dd>Most inclusive, captures all potential causal variants</dd>
<dt><strong><code>intersection</code></strong></dt>
<dd>Only variants present in credible sets from all studies</dd>
<dd>Most conservative, highest confidence variants only</dd>
<dt><strong><code>cluster</code></strong></dt>
<dd>Groups overlapping credible sets using Jaccard similarity</dd>
<dd>Balanced approach, creates meta-credible sets</dd>
</dl>
</div>
<div class="admonition note">
<p class="admonition-title">PIP Combination (<code>--combine-pip</code>)</p>
<dl>
<dt><strong><code>max</code></strong> (default)</dt>
<dd>Maximum PIP across all studies for each variant</dd>
<dd>Emphasizes strongest signals</dd>
<dt><strong><code>mean</code></strong></dt>
<dd>Average PIP across studies</dd>
<dd>Balanced view of evidence</dd>
<dt><strong><code>meta</code></strong></dt>
<dd>Meta-analysis formula: 1 - ∏(1 - PIP_i)</dd>
<dd>Accounts for independence assumption</dd>
</dl>
</div>
<h2 id="detailed-tool-descriptions">Detailed Tool Descriptions<a class="headerlink" href="#detailed-tool-descriptions" title="Permanent link">&para;</a></h2>
<h3 id="multisusie">MultiSuSiE<a class="headerlink" href="#multisusie" title="Permanent link">&para;</a></h3>
<p><strong>Best for</strong>: Multi-population analysis with shared and population-specific effects</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>mafm<span class="w"> </span>finemap<span class="w"> </span>input_loci.txt<span class="w"> </span>output/<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a><span class="w">    </span>--strategy<span class="w"> </span>multi_input<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a><span class="w">    </span>--tool<span class="w"> </span>multisusie<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a><span class="w">    </span>--max-causal<span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a><span class="w">    </span>--rho<span class="w"> </span><span class="m">0</span>.75<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a><span class="w">    </span>--scaled-prior-variance<span class="w"> </span><span class="m">0</span>.2<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a><span class="w">    </span>--pop-spec-standardization<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-7-8" name="__codelineno-7-8" href="#__codelineno-7-8"></a><span class="w">    </span>--estimate-prior-variance
</code></pre></div>
<p><strong>Key Parameters:</strong></p>
<ul>
<li><code>--rho</code>: Prior correlation between causal variants across populations (0.75)</li>
<li><code>--scaled-prior-variance</code>: Prior effect size variance (0.2)  </li>
<li><code>--pop-spec-standardization</code>: Use population-specific standardization</li>
<li><code>--estimate-prior-variance</code>: Estimate rather than fix prior variance</li>
<li><code>--pop-spec-effect-priors</code>: Population-specific effect size priors</li>
</ul>
<div class="admonition tip">
<p class="admonition-title">MultiSuSiE Guidance</p>
<ul>
<li>Higher <code>--rho</code> assumes more sharing across populations</li>
<li>Use <code>--pop-spec-standardization</code> when sample sizes vary greatly</li>
<li><code>--estimate-prior-variance</code> is usually recommended</li>
</ul>
</div>
<h3 id="susiex">SuSiEx<a class="headerlink" href="#susiex" title="Permanent link">&para;</a></h3>
<p><strong>Best for</strong>: Cross-ancestry fine-mapping with explicit modeling of population differences</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>mafm<span class="w"> </span>finemap<span class="w"> </span>input_loci.txt<span class="w"> </span>output/<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a><span class="w">    </span>--strategy<span class="w"> </span>multi_input<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a><span class="w">    </span>--tool<span class="w"> </span>susiex<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a><span class="w">    </span>--max-causal<span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a><span class="w">    </span>--mult-step<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-8-6" name="__codelineno-8-6" href="#__codelineno-8-6"></a><span class="w">    </span>--keep-ambig<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-8-7" name="__codelineno-8-7" href="#__codelineno-8-7"></a><span class="w">    </span>--min-purity<span class="w"> </span><span class="m">0</span>.5<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-8-8" name="__codelineno-8-8" href="#__codelineno-8-8"></a><span class="w">    </span>--tol<span class="w"> </span>1e-3
</code></pre></div>
<p><strong>Key Parameters:</strong></p>
<ul>
<li><code>--mult-step</code>: Use multiple refinement steps  </li>
<li><code>--keep-ambig</code>: Keep ambiguous SNPs in analysis</li>
<li><code>--min-purity</code>: Minimum purity for credible sets (0.5)</li>
<li><code>--tol</code>: Convergence tolerance (1e-3)</li>
</ul>
<h2 id="comprehensive-multi-input-example">Comprehensive Multi-Input Example<a class="headerlink" href="#comprehensive-multi-input-example" title="Permanent link">&para;</a></h2>
<p>Here's a complete workflow for multi-ancestry fine-mapping:</p>
<h3 id="step-1-input-preparation">Step 1: Input Preparation<a class="headerlink" href="#step-1-input-preparation" title="Permanent link">&para;</a></h3>
<div class="highlight"><span class="filename">multi_ancestry_loci.txt</span><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>chr<span class="w"> </span>start<span class="w">   </span>end<span class="w"> </span>popu<span class="w">    </span>sample_size<span class="w"> </span>cohort<span class="w">  </span>prefix<span class="w">  </span>locus_id
<a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a><span class="m">8</span><span class="w">   </span><span class="m">41242482</span><span class="w">    </span><span class="m">42492482</span><span class="w">    </span>AFR<span class="w"> </span><span class="m">89499</span><span class="w">   </span>MVP<span class="w"> </span>data/AFR.MVP.chr8_41242482_42492482<span class="w"> </span>chr8_41242482_42492482
<a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a><span class="m">8</span><span class="w">   </span><span class="m">41242482</span><span class="w">    </span><span class="m">42492482</span><span class="w">    </span>EUR<span class="w"> </span><span class="m">337465</span><span class="w">  </span>MVP<span class="w"> </span>data/EUR.MVP.chr8_41242482_42492482<span class="w"> </span>chr8_41242482_42492482
<a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a><span class="m">8</span><span class="w">   </span><span class="m">41242482</span><span class="w">    </span><span class="m">42492482</span><span class="w">    </span>EUR<span class="w"> </span><span class="m">442817</span><span class="w">  </span>UKBB<span class="w">    </span>data/EUR.UKBB.chr8_41242482_42492482<span class="w">    </span>chr8_41242482_42492482
<a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a><span class="m">8</span><span class="w">   </span><span class="m">41242482</span><span class="w">    </span><span class="m">42492482</span><span class="w">    </span>SAS<span class="w"> </span><span class="m">8253</span><span class="w">    </span>UKBB<span class="w">    </span>data/SAS.UKBB.chr8_41242482_42492482<span class="w">    </span>chr8_41242482_42492482
</code></pre></div>
<h3 id="step-2-quality-control">Step 2: Quality Control<a class="headerlink" href="#step-2-quality-control" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="c1"># Run comprehensive QC</span>
<a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a>mafm<span class="w"> </span>qc<span class="w"> </span>multi_ancestry_loci.txt<span class="w"> </span>qc_results/<span class="w"> </span>--threads<span class="w"> </span><span class="m">4</span>
<a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a>
<a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a><span class="c1"># Review heterogeneity</span>
<a id="__codelineno-10-5" name="__codelineno-10-5" href="#__codelineno-10-5"></a>head<span class="w"> </span>qc_results/cochran_q.txt
</code></pre></div>
<h3 id="step-3a-cross-ancestry-analysis-with-multisusie">Step 3A: Cross-Ancestry Analysis with MultiSuSiE<a class="headerlink" href="#step-3a-cross-ancestry-analysis-with-multisusie" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="c1"># Meta-analyze across all populations</span>
<a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a>mafm<span class="w"> </span>pipeline<span class="w"> </span>multi_ancestry_loci.txt<span class="w"> </span>results_cross_ancestry/<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a><span class="w">    </span>--meta-method<span class="w"> </span>meta_all<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a><span class="w">    </span>--strategy<span class="w"> </span>multi_input<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a><span class="w">    </span>--tool<span class="w"> </span>multisusie<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-11-6" name="__codelineno-11-6" href="#__codelineno-11-6"></a><span class="w">    </span>--max-causal<span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="se">\</span>
<a id="__codelineno-11-7" name="__codelineno-11-7" href="#__codelineno-11-7"></a><span class="w">    </span>--rho<span class="w"> </span><span class="m">0</span>.8<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-11-8" name="__codelineno-11-8" href="#__codelineno-11-8"></a><span class="w">    </span>--pop-spec-standardization<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-11-9" name="__codelineno-11-9" href="#__codelineno-11-9"></a><span class="w">    </span>--estimate-prior-variance
</code></pre></div>
<h3 id="step-3b-population-specific-analysis">Step 3B: Population-Specific Analysis<a class="headerlink" href="#step-3b-population-specific-analysis" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="c1"># Meta-analyze within populations, then combine</span>
<a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a>mafm<span class="w"> </span>pipeline<span class="w"> </span>multi_ancestry_loci.txt<span class="w"> </span>results_pop_specific/<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-12-3" name="__codelineno-12-3" href="#__codelineno-12-3"></a><span class="w">    </span>--meta-method<span class="w"> </span>meta_by_population<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-12-4" name="__codelineno-12-4" href="#__codelineno-12-4"></a><span class="w">    </span>--strategy<span class="w"> </span>post_hoc_combine<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-12-5" name="__codelineno-12-5" href="#__codelineno-12-5"></a><span class="w">    </span>--tool<span class="w"> </span>susie<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-12-6" name="__codelineno-12-6" href="#__codelineno-12-6"></a><span class="w">    </span>--combine-cred<span class="w"> </span>cluster<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-12-7" name="__codelineno-12-7" href="#__codelineno-12-7"></a><span class="w">    </span>--combine-pip<span class="w"> </span>meta<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-12-8" name="__codelineno-12-8" href="#__codelineno-12-8"></a><span class="w">    </span>--jaccard-threshold<span class="w"> </span><span class="m">0</span>.2
</code></pre></div>
<h3 id="step-4-compare-approaches">Step 4: Compare Approaches<a class="headerlink" href="#step-4-compare-approaches" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="c1"># Compare results from different strategies</span>
<a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a><span class="kn">import</span><span class="w"> </span><span class="nn">json</span>
<a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a>
<a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a><span class="c1"># Load results</span>
<a id="__codelineno-13-6" name="__codelineno-13-6" href="#__codelineno-13-6"></a><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;results_cross_ancestry/creds.json&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
<a id="__codelineno-13-7" name="__codelineno-13-7" href="#__codelineno-13-7"></a>    <span class="n">cross_ancestry</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
<a id="__codelineno-13-8" name="__codelineno-13-8" href="#__codelineno-13-8"></a>
<a id="__codelineno-13-9" name="__codelineno-13-9" href="#__codelineno-13-9"></a><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;results_pop_specific/creds.json&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
<a id="__codelineno-13-10" name="__codelineno-13-10" href="#__codelineno-13-10"></a>    <span class="n">pop_specific</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
<a id="__codelineno-13-11" name="__codelineno-13-11" href="#__codelineno-13-11"></a>
<a id="__codelineno-13-12" name="__codelineno-13-12" href="#__codelineno-13-12"></a><span class="c1"># Compare credible sets and PIPs</span>
<a id="__codelineno-13-13" name="__codelineno-13-13" href="#__codelineno-13-13"></a><span class="n">cross_pips</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;results_cross_ancestry/pips.txt&#39;</span><span class="p">,</span> 
<a id="__codelineno-13-14" name="__codelineno-13-14" href="#__codelineno-13-14"></a>                        <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;SNP&#39;</span><span class="p">,</span> <span class="s1">&#39;PIP&#39;</span><span class="p">])</span>
<a id="__codelineno-13-15" name="__codelineno-13-15" href="#__codelineno-13-15"></a><span class="n">pop_pips</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;results_pop_specific/pips.txt&#39;</span><span class="p">,</span> 
<a id="__codelineno-13-16" name="__codelineno-13-16" href="#__codelineno-13-16"></a>                      <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;SNP&#39;</span><span class="p">,</span> <span class="s1">&#39;PIP&#39;</span><span class="p">])</span>
</code></pre></div>
<h2 id="advanced-multi-input-considerations">Advanced Multi-Input Considerations<a class="headerlink" href="#advanced-multi-input-considerations" title="Permanent link">&para;</a></h2>
<h3 id="handling-missing-data">Handling Missing Data<a class="headerlink" href="#handling-missing-data" title="Permanent link">&para;</a></h3>
<p>When variants are missing in some populations:</p>
<div class="admonition warning">
<p class="admonition-title">Missing Data Strategies</p>
<dl>
<dt><strong>Complete case analysis</strong></dt>
<dd>Only use variants present in all studies</dd>
<dd>Reduces power but ensures consistency</dd>
<dt><strong>Imputation</strong></dt>
<dd>Impute missing variants using population-specific references</dd>
<dd>Requires careful validation</dd>
<dt><strong>Weighted analysis</strong></dt>
<dd>Weight contributions by data availability</dd>
<dd>Built into MAFM meta-analysis</dd>
</dl>
</div>
<h3 id="population-stratification">Population Stratification<a class="headerlink" href="#population-stratification" title="Permanent link">&para;</a></h3>
<p>When populations have internal structure:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a><span class="c1"># Use more conservative thresholds</span>
<a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a>mafm<span class="w"> </span>finemap<span class="w"> </span>input.txt<span class="w"> </span>output/<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-14-3" name="__codelineno-14-3" href="#__codelineno-14-3"></a><span class="w">    </span>--strategy<span class="w"> </span>multi_input<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-14-4" name="__codelineno-14-4" href="#__codelineno-14-4"></a><span class="w">    </span>--tool<span class="w"> </span>multisusie<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-14-5" name="__codelineno-14-5" href="#__codelineno-14-5"></a><span class="w">    </span>--rho<span class="w"> </span><span class="m">0</span>.5<span class="w"> </span><span class="se">\</span>
<a id="__codelineno-14-6" name="__codelineno-14-6" href="#__codelineno-14-6"></a><span class="w">    </span>--min-abs-corr<span class="w"> </span><span class="m">0</span>.8
</code></pre></div>
<h3 id="computational-considerations">Computational Considerations<a class="headerlink" href="#computational-considerations" title="Permanent link">&para;</a></h3>
<p>Large multi-population analyses can be computationally intensive:</p>
<div class="admonition tip">
<p class="admonition-title">Performance Optimization</p>
<ul>
<li>Use <code>--threads</code> for parallelization</li>
<li>Consider analyzing loci separately for very large studies</li>
<li>Monitor memory usage with many populations</li>
<li>Use post-hoc combination for maximum flexibility</li>
</ul>
</div>
<h2 id="interpreting-multi-input-results">Interpreting Multi-Input Results<a class="headerlink" href="#interpreting-multi-input-results" title="Permanent link">&para;</a></h2>
<h3 id="population-specific-effects">Population-Specific Effects<a class="headerlink" href="#population-specific-effects" title="Permanent link">&para;</a></h3>
<p>Look for variants with:
- High PIPs in some populations but not others
- Different credible sets across populations
- High Cochran's Q values</p>
<h3 id="shared-causal-variants">Shared Causal Variants<a class="headerlink" href="#shared-causal-variants" title="Permanent link">&para;</a></h3>
<p>Evidence includes:
- Consistent PIPs across populations
- Overlapping credible sets
- Low heterogeneity in meta-analysis</p>
<h3 id="meta-analysis-benefits">Meta-Analysis Benefits<a class="headerlink" href="#meta-analysis-benefits" title="Permanent link">&para;</a></h3>
<p>Compare single-population vs. meta-analysis:
- Increased resolution (smaller credible sets)
- Higher PIPs for true causal variants
- Discovery of additional signals</p>
<h2 id="common-multi-input-issues">Common Multi-Input Issues<a class="headerlink" href="#common-multi-input-issues" title="Permanent link">&para;</a></h2>
<div class="admonition warning">
<p class="admonition-title">Troubleshooting Multi-Input Analysis</p>
<dl>
<dt><strong>Excessive heterogeneity</strong></dt>
<dd>Use <code>meta_by_population</code> instead of <code>meta_all</code></dd>
<dd>Check for technical differences between studies</dd>
<dd>Consider population-specific analyses</dd>
<dt><strong>No shared signals</strong></dt>
<dd>May indicate true population-specific effects</dd>
<dd>Check LD structure differences</dd>
<dd>Verify consistent variant coding</dd>
<dt><strong>Computational issues</strong></dt>
<dd>Reduce number of populations analyzed together</dd>
<dd>Use post-hoc combination strategy</dd>
<dd>Increase available memory/compute resources</dd>
</dl>
</div>
<h2 id="next-steps">Next Steps<a class="headerlink" href="#next-steps" title="Permanent link">&para;</a></h2>
<ul>
<li><strong><a href="../advanced/">Advanced Topics</a></strong> - Deep dive into parameter optimization and custom workflows</li>
<li>Review specific tool documentation for detailed parameter guidance</li>
<li>Consider population genetics factors in result interpretation </li>
</ul>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.indexes", "navigation.instant", "navigation.tabs.sticky", "navigation.sections", "navigation.expand", "toc.follow"], "search": "../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.13a4f30d.min.js"></script>
      
    
  </body>
</html>